app/models/concerns/taxonomix.rb

```diff
@@ -12,6 +12,7 @@ module Taxonomix
              :through => TAXONOMY_JOIN_TABLE, :source => :taxonomy,
              :validate => false
     after_initialize :set_current_taxonomy
+    before_save :add_taxonomies_having_select_all_enabled
```

Comment:
Why do this every save and not only on create?
Isn't it already saved by the taxonomy when it's saved for all existing taxable taxonomies?

Reply:
I suggested this in https://github.com/theforeman/foreman/pull/4111#discussion_r128496408, the other option is to disable this from hammer/API. We could add a validation check and disable update if some of linked taxonomies "all" has been checked. But we'd need to load all taxonomies in this case too. If there's no change we don't fire any extra SQL call. Maybe the performance could be slightly improved by using `organization_ids=` instead of `organizations=` but we need to load them anyway.

Reply:
Currently I am using `organizations=` only, I will try to modify it to `organization_ids=`  and will check if I can minimize any SQL call.

Reply:
@ares and @orrabin,

I have now used `organization_ids=` instead of `organizations=`.
Previously I thought that disabling update from hammer/API will be good to have to make data consistent. To add this change, we need to load taxonomies again like you mentioned. 

So I will prefer to go with update on every save. I have added check that if taxonomies are not already assigned to an object in that case only update will happen for taxonomies.
And I think current code changes will take care of data consistency.

Please let me know if I misunderstood anything.